qiskit_experiments/framework/composite/batch_experiment.py

```diff
@@ -13,62 +13,80 @@
 Batch Experiment class.
 """
 
-from collections import OrderedDict
+from typing import List
 
 from qiskit import QuantumCircuit
 
+
 from .composite_experiment import CompositeExperiment
 
 
 class BatchExperiment(CompositeExperiment):
-    """Batch experiment class"""
+    """Batch experiment class.
+
+    This experiment takes multiple experiment instances and generates
+    a list of flattened circuit to execute.
+    The experimental circuits are executed ony be one on the target backend as a single job.
+
+    If an experiment analysis needs results of different types of experiments,
+    ``BatchExperiment`` may be convenient to describe the flow of the entire experiment.
+
+    The experimental result of ``i``-th experiment can be accessed by
+
+    .. code-block:: python3
+
+        batch_exp = BatchExperiment([exp1, exp2, exp3])
+        batch_result = batch_exp.run(backend)
+
+        exp1_res = batch_result.component_experiment_data(0)  # data of exp1
+        exp2_res = batch_result.component_experiment_data(1)  # data of exp2
+        exp3_res = batch_result.component_experiment_data(2)  # data of exp3
+
+    One can also create a custom analysis class that estimates some parameters by
+    combining above analysis results. Here the ``exp*_res`` is a single
+    :py:class:`~qiskit_experiments.framework.experiment_data.ExperimentData` class of
+    a standard experiment, and the associated analysis will be performed once the batch job
+    is completed. Thus analyzed parameter value of each experiment can be obtained as usual.
+
+    .. code-block:: python3
+
+        param_x = exp1_res.analysis_results("target_parameter_x")
+        param_y = exp2_res.analysis_results("target_parameter_y")
+        param_z = exp3_res.analysis_results("target_parameter_z")
+
+        param_xyz = param_x + param_y + param_z  # do some computation
+
+    The final parameter ``param_xyz`` can be returned as an outcome of this batch experiment.
+    """
 
     def __init__(self, experiments):
         """Initialize a batch experiment.
 
         Args:
             experiments (List[BaseExperiment]): a list of experiments.
         """
-
-        # Generate qubit map
-        self._qubit_map = OrderedDict()
-        logical_qubit = 0
-        for expr in experiments:
-            for physical_qubit in expr.physical_qubits:
-                if physical_qubit not in self._qubit_map:
-                    self._qubit_map[physical_qubit] = logical_qubit
-                    logical_qubit += 1
-        qubits = tuple(self._qubit_map.keys())
+        qubits = sorted(set(sum([list(expr.physical_qubits) for expr in experiments], [])))
         super().__init__(experiments, qubits)
 
-    def circuits(self, backend=None):
+    def _flatten_circuits(
```

Comment:
Why not keeping the name `circuits` ? What if we'll have a new type of a composite experiment subclass in which we the circuit generation is not really flattening or concatenation ? Maybe some variational algorithm for example. I think `circuits` is a proper general term. Just a thought...

Reply:
Oh this is due to poor gitdiff. The `circuits` method still exists but not used because we always get circuit from nested experiments.
https://github.com/Qiskit/qiskit-experiments/blob/d5ca18d207a6deb6ed690b6ebdc68c2ed8b5246b/qiskit_experiments/framework/composite/composite_experiment.py#L130-L143

Actually this is one of my concerns in this PR. Previously, this `circuits` method provides a logic to combine circuits without transpiling. However, this doesn't allow us to combine different types of experiments, e.g. RB and calibration, due to different transpilation behavior.

I think an inner-loop of a variational algorithm can be implemented as a `BaseExperiment` rather than composite. The loop type experiment should be different base class because these base classes don't implements a logic for iteration.